src/v0/destinations/criteo_audience/transform.js

```diff
@@ -0,0 +1,108 @@
+const {
+  BASE_ENDPOINT,
+  CRITEO_ADD_USER,
+  CRITEO_REMOVE_USER
+} = require("./config");
+const {
+  defaultRequestConfig,
+  removeUndefinedAndNullValues,
+  simpleProcessRouterDest,
+  defaultPatchRequestConfig
+} = require("../../util");
+const { InstrumentationError } = require("../../util/errorTypes");
+
+const { getAccessToken, preparePayload } = require("./util");
+
+const prepareResponse = (payload, audienceId, accessToken) => {
+  const response = defaultRequestConfig();
+  response.endpoint = `${BASE_ENDPOINT}audiences/${audienceId}/contactlist`;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  response.method = defaultPatchRequestConfig.requestMethod;
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json",
+    Accept: "application/json"
+  };
+  return response;
+};
+/**
+ * This function is used for building the final response to be returned.
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const responseBuilder = async (message, destination, accessToken) => {
+  const responseArray = [];
+  let criteoAddPayload = {};
+  let criteoRemovePayload = {};
+  const { Config } = destination;
+  const { audienceId } = Config;
+  const { listData } = message.properties;
+
+  if (listData[CRITEO_ADD_USER]) {
+    criteoAddPayload = preparePayload(
+      listData[CRITEO_ADD_USER],
+      CRITEO_ADD_USER,
+      Config
+    );
+    responseArray.push(
+      prepareResponse(criteoAddPayload, audienceId, accessToken)
+    );
+  }
+  if (listData[CRITEO_REMOVE_USER]) {
+    criteoRemovePayload = preparePayload(
+      listData[CRITEO_REMOVE_USER],
+      CRITEO_REMOVE_USER,
+      Config
+    );
+    responseArray.push(
+      prepareResponse(criteoRemovePayload, audienceId, accessToken)
+    );
+  }
```

Comment:
It is repetitive. Can't we use any loop here?

Reply:
Yes Will update

Reply:
Done